authorization-server/src/modules/auth/guards/valid-refresh-token.guard.ts

```diff
@@ -9,12 +16,41 @@ export class ValidRefreshTokenGuard implements CanActivate {
   ) {}
 
   async canActivate(context: ExecutionContext): Promise<boolean> {
-    const { body, cookies } = context.switchToHttp().getRequest();
+    const request: Request = context.switchToHttp().getRequest();
+    const { body, cookies, method } = request;
+
+    let refreshToken: string | undefined;
+
+    if (method === 'POST') {
+      refreshToken = this.extractFromPostBodyOrCookies(body, cookies);
+    } else if (method === 'GET') {
+      refreshToken = this.extractFromQueryStringOrCookies(
+        request.cookies,
+        request.query as { refresh_token?: string },
+      );
```

Comment:
Looking at now, I think this single guard approach is not ideal because it would need to change if a different request method was added. I think it might be a violation of the "open/closed principle". Ideally, the class wouldn't need to change to be used in a different context.
As I'm not sure what the best way to improve it is and as we're unlikely to have another method in the future, I think it's okay as is. But maybe something to keep in mind in the future.

Reply:
I agree - this is not the cleanest, but I can't see a more readable approach here.